# HG changeset patch
# User Rik <rik@octave.org>
# Date 1475528878 25200
#      Mon Oct 03 14:07:58 2016 -0700
# Branch stable
# Node ID 2bca08793651b20aef14a34c07f9dc72bb8fe786
# Parent  43fe18c1129e0cd0496a55c4ecb3bfa71a9252db
Fix regression with '@' character in gnuplot text fields (bug #49260).

* __gnuplot_draw_axes__.m: Change regexprep replacement pattern from
'\@' to '\\@' since Matlab requires one round of interpretation despite
the single quotes.

diff --git a/scripts/plot/util/private/__gnuplot_draw_axes__.m b/scripts/plot/util/private/__gnuplot_draw_axes__.m
--- a/scripts/plot/util/private/__gnuplot_draw_axes__.m
+++ b/scripts/plot/util/private/__gnuplot_draw_axes__.m
@@ -2411,17 +2411,17 @@ function [str, f, s] = __maybe_munge_tex
       if (isnumeric (str{n}))
         str{n} = num2str (str{n});
       endif
     endfor
     str = sprintf ("%s\n", str{:})(1:end-1);
   endif
 
   if (enhanced)
-    str = regexprep (str, '(?<!\\)@', '\@');
+    str = regexprep (str, '(?<!\\)@', '\\@');
   endif
 
   if (enhanced)
     if (strcmp (obj.interpreter, "tex"))
       if (iscellstr (str))
         for n = 1:numel (str)
           str{n} = __tex2enhanced__ (str{n}, fnt, it, bld);
         endfor
